cache/manager_test.go

```diff
@@ -1542,12 +1546,17 @@ func TestConversion(t *testing.T) {
 
 	// Tests all combination of the conversions from type i to type j preserve
 	// the uncompressed digest.
-	allCompression := []compression.Type{compression.Uncompressed, compression.Gzip, compression.EStargz, compression.Zstd}
+	allCompression := []compression.Type{compression.Uncompressed, compression.Gzip, compression.EStargz, compression.Zstd, compression.Nydus}
 	eg, egctx := errgroup.WithContext(ctx)
-	for _, orgDesc := range []ocispecs.Descriptor{orgDescGo, orgDescSys} {
+	for k, orgDesc := range []ocispecs.Descriptor{orgDescGo, orgDescSys} {
 		for _, i := range allCompression {
 			compSrc := compression.New(i)
 			for _, j := range allCompression {
+				// Currently the nydus compression type only ensures the consistency of archive/tar
+				// implementation in Go, so need to skip the system tar command case for it.
+				if k == 1 && (i == compression.Nydus || j == compression.Nydus) {
+					continue
+				}
```

Comment:
It looks like that this breaks the cache if a conversion from a layer which is tar-ed by non-go implementation into nydus then later another conversion happens from that nydus blob to another compression.

Reply:
Thanks for the reply! I want to explain that nydus is not an archive format, if we convert a system tar to nydus and then convert nydus back to tar, although nydus can guarantee filesystem consistency, but it can’t do a complete restore of the system tar, it will change its file entry path, for example, `./foo1` -> `foo1` (it lost `./` for file path in the tar header), but nydus makes sure that the decompressed tar is as consistent as possible with the OCI tar format (in go implementation), and does strictly guarantee in filesystem consistency. So can we consider skipping the hash check test case between nydus and system tar first?